legend-engine-xts-authentication/legend-engine-xt-authentication-connection-factory/src/main/java/org/finos/legend/connection/CredentialExtractor.java

```diff
@@ -20,6 +20,8 @@
 
 import java.util.Optional;
 
+// phase 0 - rename SPEC to CONFIG ?
+// phase 1 - Can CredentialBuilder and CredentialExtractor be unified ? Why do we need two concepts ?
```

Comment:
Certainly can, I think it's clearer to define a special kind of credential builder that does not really do anything except for extracting the credential from the identity. I don't think we want to support:

`PlaintextPasswordCredential --> PlaintextPasswordCredential`, but for cases like `Kerberos` we do